### ISSUE TITLE
Invalid or zero field values ConsumptionProbe

### ISSUE THREAD

**Issue by sedran** [vladimir-bukhtoyarov/bucket4j#71]
Hello,

I want to provide response headers to my API clients such as number of remaining tokens (X-RateLimit-Remaining) and time to refill the bucket (X-RateLimit-Remaining) however;

- When `consumed` is true, `nanosToWaitForRefill` is zero
- When `FixedRefillInterval` is set on Bandwidth, `nanosToWaitForRefill` is not effected by this setting.

For example, I have created a refill of 60 seconds and a bucket with 10 tokens. I've also set FixedRefillInterval as 60 seconds. When the client goes out of tokens in the first second, nanosToWaitForRefill is returned as 6 seconds (60 /  10) instead of 60 (fixed refill interval)

- Without `FixedRefillInterval`, it seems like `nanosToWaitForRefill` does not reduce. Until a new token is created, I always get a value of 6 seconds (60 / 10). I expect it to reduce at each call, like 6 seconds on first call when the bucket is empty, then after 1 second, it should return 5 seconds, etc. Am I thinking wrong?

I've also put an example from my code:

```
@Component
public class Bucket4jRateLimitTracker implements RateLimitTracker, InitializingBean {
	private static final Logger LOG = LoggerFactory.getLogger(Bucket4jRateLimitTracker.class);

	@Autowired
	private HazelcastInstance hz;

	@Autowired
	private RateLimitProvider rateLimitProvider;

	private IMap<RateLimitMapKey, GridBucketState> gridBucketStateMap;
	private ProxyManager<RateLimitMapKey> buckets;

	@Override
	public void afterPropertiesSet() {
		gridBucketStateMap = hz.getMap(Constants.HZ_THROTTLING_MAP);
		buckets = Bucket4j.extension(Hazelcast.class).proxyManagerForMap(gridBucketStateMap);
	}

	@Override
	public RateLimitWindow tryConsume(RateLimitMapKey rateLimitMapKey) {
		final RateLimit rateLimit = rateLimitProvider.getRateLimit(rateLimitMapKey);
		if (!rateLimit.isLimited()) {
			return UnlimitedRateLimitWindow.instance();
		}

		final Supplier<BucketConfiguration> configurationLazySupplier = getConfigSupplier(rateLimit);
		final Bucket bucket = buckets.getProxy(rateLimitMapKey, configurationLazySupplier);

		final ConsumptionProbe probe = bucket.tryConsumeAndReturnRemaining(1);
		return new Bucket4JRateLimitWindow(probe);
	}

	@Override
	public void updateBucket(RateLimitMapKey rateLimitMapKey, RateLimit rateLimit) {
		final Optional<Bucket> bucketOptional = buckets.getProxy(rateLimitMapKey);
		bucketOptional.ifPresent(bucket -> {
			LOG.debug("Executing replace bucket for {} = {}", rateLimitMapKey, rateLimit);
			bucket.replaceConfiguration(getConfig(rateLimit));
		});
	}

	@Override
	public void deleteBucket(RateLimitMapKey rateLimitMapKey) {
		LOG.debug("deleteBucket = {}", rateLimitMapKey);
		gridBucketStateMap.remove(rateLimitMapKey);
	}

	private Supplier<BucketConfiguration> getConfigSupplier(RateLimit rateLimit) {
		return () -> getConfig(rateLimit);
	}

	private BucketConfiguration getConfig(RateLimit rateLimit) {
		final Duration refillPeriod = rateLimit.getPeriod().asDuration();
		final Refill refill = Refill.of(rateLimit.getAmount(), refillPeriod);
		return Bucket4j.configurationBuilder() //
				.addLimit(Bandwidth.classic(rateLimit.getAmount(), refill).withFixedRefillInterval(refillPeriod)) //
				.build();
	}
}
```

**Comment by vladimir-bukhtoyarov** [comment]
@sedran please read [javadocs for ConsumptionProbe#getNanosToWaitForRefill](https://github.com/vladimir-bukhtoyarov/bucket4j/blob/master/bucket4j-core/src/main/java/io/github/bucket4j/ConsumptionProbe.java#L68). I have nothing else to add, javadacs explicitly answer to your question.

**Comment by vladimir-bukhtoyarov** [comment]
> It is a error, I will fix it in 4.0.1

Fixed and deployed.
